src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -2566,32 +2566,44 @@ public int ChildCount()
         }
 
         // Determines whether the node supports a compilation / code generation strategy based on walking the node tree.
-        internal bool SupportsCompilation()
+        // Also returns a human-readable string to explain the reason to the source generator.
+        internal bool SupportsCompilation([NotNullWhen(false)] out string? reason)
         {
```

Comment:
We should add a comment that the text reasons are used only for the source generation and do not need to be localized.